half-lajf-app/src/components/layout/Sidenav.js

```diff
@@ -1,27 +1,44 @@
 import React, { Component } from 'react'
-import { Link } from 'react-router-dom'
+import { NavLink } from 'react-router-dom'
 import M from "materialize-css"
+import './Sidenav.css'
 
 class Sidenav extends Component {
   componentDidMount() {
     var elem = document.querySelector(".sidenav");
-    var instance = M.Sidenav.init(elem, {
+    M.Sidenav.init(elem, {
         edge: "right",
     });
+    var elems = document.querySelector('.collapsible');
+    M.Collapsible.init(elems, {
+      inDuration: 380,
+    });
   }
-
   render() {
     return (
       <div>
         <ul id="mobile-demo" className="sidenav">
-          <li><Link to='/'>Categories</Link></li>
-          <li><Link to='/'>Random meme</Link></li>
-          <li><Link to='/'>Sign up</Link></li>
-          <li><Link to='/'>Sign in</Link></li>
-          <li><Link to='/'>Your profile</Link></li>
-          <li><Link to='/' class="red-text">Sign out</Link></li>
+          <li>
+            <ul className="collapsible">
+              <li>
+                <div className="collapsible-header black-text"><i className="material-icons collapsible-secondary">arrow_drop_down</i><b>Categories</b></div>
+                <div className="collapsible-body black-text">
+                  <a href='#' className="black-text sidenav-close"><span style={{marginLeft: 16}}>Category</span></a>
+                  <a href='#' className="black-text sidenav-close"><span style={{marginLeft: 16}}>Category</span></a>
+                  <a href='#' className="black-text sidenav-close"><span style={{marginLeft: 16}}>Category</span></a>
+                  <a href='#' className="black-text sidenav-close"><span style={{marginLeft: 16}}>Category</span></a>
+                  <a href='#' className="black-text sidenav-close"><span style={{marginLeft: 16}}>Category</span></a>
```

Comment:
Hardcoded value. What about adding new categories in the future?

Reply:
Don't worry, I'm gonna fix it soon by commiting my fist to your face :)